index.js

```diff
@@ -17,7 +17,12 @@ function (createConnection) {
       //use "connection" to match core (net) api.
       emitter.on('connection', onConnect)
 
-    var backoffMethod = (backoff[opts.type] || backoff.fibonacci) (opts)
+    var backoffStrategy = opts.strategy || opts.type
+    var backoffMethod
+    if (backoffStrategy instanceof backoff.Backoff)
```

Comment:
instanceof checks aren't a good idea with npm as you can't be sure this module gets the same version of backoff as the module consuming it does. i'd just check if it's a string, otherwise if it's an object it's a strategy